MetaMask/metamask-extension
# fix: trying to access an undefined object in swaps review quote compo…

## **Description**

Fixes the two errors with trade and decimals being undefined that have been causing crashes starting around the 12.1/12.2 release. I was unable to find the root cause of this issue. Variables in the redux store seem to return as undefined, which leads me to think it might be some sort of redux race condition. The lowest common denominator of this error seems to be that `getUsedQuote` selector in the `ReviewQuote` component. I have added an additional condition to the render guard in the parent component `prepare-swaps-page` to prevent these errors


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27708?quickstart=1)

## **Related issues**

[MMS-1569](https://consensyssoftware.atlassian.net/jira/software/projects/MMS/boards/447/backlog?assignee=5ae37c7e42b8a62c4e15d92a&selectedIssue=MMS-1569)

## **Manual testing steps**

1. Open Swaps Page
2. Enter swap amount
3. Edit to token and amount rapidly multiple times 
4. Page should not crash

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[MMS-1569]: https://consensyssoftware.atlassian.net/browse/MMS-1569?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
